internal/pkg/manifest/pipeline.go

```diff
@@ -188,10 +188,12 @@ type Build struct {
 
 // PipelineStage represents a stage in the pipeline manifest
 type PipelineStage struct {
-	Name             string      `yaml:"name"`
-	RequiresApproval bool        `yaml:"requires_approval,omitempty"`
-	TestCommands     []string    `yaml:"test_commands,omitempty"`
-	Deployments      Deployments `yaml:"deployments,omitempty"`
+	Name             string             `yaml:"name"`
+	RequiresApproval bool               `yaml:"requires_approval,omitempty"`
+	TestCommands     []string           `yaml:"test_commands,omitempty"`
+	Deployments      Deployments        `yaml:"deployments,omitempty"`
+	PreDeployment    PrePostDeployments `yaml:"pre_deployment,omitempty"`
```

Comment:
There can be multiple pre-deployment actions, hence plural! In addition, "deployments" field is plural as well.
```suggestion
	PreDeployment    PrePostDeployments `yaml:"pre_deployments,omitempty"`
```

Reply:
Yeah, I went back and forth on that. I had settled on singular because the multiple possible pre-deployment actions are all within a singular deployment category, and same for post-, but yes, I can see the other side, too. And consistency with `deployments` is nice. Will change.